test/controllers/api/v2/smart_variables_controller_test.rb

```diff
@@ -102,4 +102,46 @@ class Api::V2::SmartVariablesControllerTest < ActionController::TestCase
       assert_equal parameter.hidden_value, show_response['default_value']
     end
   end
+
+  test "should accept boolean that is not sent as string" do
```

Comment:
All three files here have 42 new lines of testing the same thing, the logic of `cast_value`. Once you extract that logic to a common place (app/services sounds fine but it's your choice), just test that logic somewhere *once*, and use `use_before_action` in the controller tests to ensure the controllers run it before the defined actions https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/action_controller/callback_matcher.rb#L54